### PR TITLE
Take managed clusters and managed rules into account

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
I apologize for a PR this big, I won't do that again in the future (for everyone's sake...)

New features/bugfixes:

- all relevant v2 endpoints take managed clusters (has managed flag from AMS API) and managed rules (have `osd_customer` tag) into account. Managed cluster must only show managed rules. Non-managed clusters are showing all rules like before
    - `ReportEndpointV2 = "cluster/{cluster}/reports"`
    - `ClustersDetail = "rule/{rule_selector}/clusters_detail"`
    - `ClustersRecommendationsEndpoint = "clusters"`
    - `RecommendationsListEndpoint = "rule"`
- `org_overview` takes managed clusters are rules into account
- `org_overview` is now consistent with other "overview" endpoints (such as `/clusters`)
- `org_overview` now also takes both rule acks and rule disables into account

Fixes CCXDEV-7235 CCXDEV-7387

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)

## Testing steps
final tests against production AMS API and modified rule hits to include managed/non-managed rules

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: content/content.go [hunk 1]
```diff
@@ -483,7 +513,7 @@ func FetchRuleContent(rule ctypes.RuleOnReport, OSDEligible bool) (
 		return
 	}
 
-	if OSDEligible && !ruleWithContent.NotRequireAdmin {
+	if OSDEligible && !ruleWithContent.OSDCustomer {
```
Comment (tisnik) [inline content/content.go:516]: like it, it's much more specific

**Review by dpensi — APPROVED** [review]

**Review by tisnik — APPROVED** [review]
It looks ok!

File: server/server.go [hunk 2]
```diff
@@ -1400,3 +1404,49 @@ func (server *HTTPServer) readListOfDisabledRulesForClusters(
 
 	return response.DisabledRules, nil
 }
+
+// Method readListOfClusterDisabledRules returns user disabled rules for given cluster list
```
Comment (joselsegura) [inline server/server.go:1408]: Mismatch between the comment and the method name

**Review by joselsegura — APPROVED** [review]
Just a nitpick. What a great job!

**Review by juandspy — APPROVED** [review]

File: server/server.go (hunk 2 above)
Comment (Bee-lee) [reply server/server.go]: thank you, good catch

**Pull request merged**